fix(core): add toString implementation to signals

crisbeto opened the pull request:
Since signals are functions, currently stringifying them reveals the implementation of the function. This can lead to confusion since it contains internal implementation details. These changes add static `toString` function to address the issue.

**Note:** it's tempting to have `toString` output the actual value of the signal, but that would encourage users not to call the function which will be problematic in the long run. That's why these changes are using a static string instead.

devversion commented on packages/core/src/authoring/input_signal.ts:
```diff
@@ -99,5 +102,6 @@ export function createInputSignal<ReadT, WriteT>(
   }
 
   (inputValueFn as any)[SIGNAL] = node;
+  inputValueFn.toString = signalInputToString;
```
Could that be part directly of the `INPUT_SIGNAL_NODE`? (where we have all the initial values?)

crisbeto replied:
I'd have to debug it a bit, but I tried it quickly and it didn't get picked up.

devversion replied:
Would be nice to look into it so that this can live in the reactive node blueprint code (since it's always the same value anyway per reactive node)

crisbeto replied:
I got some time to revisit this. So the problem with putting it into the signal node is that the getter function doesn't inherit from it, it gets patched onto the getter function. I could change it to something like `getter.toString = node.toString` but I'm not sure that's much better.

devversion replied:
Fair enough. I think we can leave it as is then.

pkozlowski-opensource replied:
Ha ha - I was actually also playing with the ideas of putting the actual logic on a node and then having getter's toString delegate to the node. But agree with @crisbeto that it doesn't buy us the whole lot so +1 for leaving it as-is.

devversion replied:
Yeah, this is fine. Was mostly curious, if it's possible.

eneajaho commented:
Isn't this method a great use for the dev tools feature where we can see the Signals graph?

crisbeto commented:
I'm not sure if it's worth shipping the extra code for traversing the graph, given that the same information can be displayed better in the dev tools.

pkozlowski-opensource reviewed (approved):
LGTM. But now we've got also query signals so you might want to include those as well - or I can do it in the follow-up PR.

crisbeto committed:
fix(core): add toString implementation to signals

Since signals are function, currently stringifying them reveals the implementation of the function. This can lead to confusion since it contains internal implementation details. These changes add static `toString` function to address the issue.

**Note:** it's tempting to have `toString` output the actual value of the signal, but that would encourage users not to call the function which will be problematic in the long run. That's why these changes are using a static string instead.

crisbeto commented:
Updated with a `toString` implementation for query signals.

jessicajaniuk commented:
This PR was merged into the repository by commit 656bc282e345c5e37a9189a0a4daa631e02c31bf.
